backend/firebaseFunctions.js

```diff
@@ -70,28 +70,76 @@ export async function createCircle(data) {
 	data.admin[`${UID}`] = true;
 	data.owner = UID;
 
+	let circles = Object.keys((await readData(`prayer_circle/circles`)) || {});
+	const allAdminCodes = new Array(circles.length);
+	const allPublicCodes = new Array(circles.length);
+	for (let i = 0; i < circles.length; i++) {
+		let circle = circles[i];
+		let circleData =
+			(await readData(`prayer_circle/circles/${circle}`)) || {};
```

Comment:
make sure to update the dbstructure.json file with your new changes. Here, i would suggest that in a new circle you store
`codes: {
    admin: 12345678,
    public: 12345678
}`

this way, you can use `readData(`prayer_circle/circles/${circle}/codes`)`. doing this will make the request smaller, reduce processing, and also make help keep the structure clean